{"name": "Test job countries from the anahr website against Peviitor API Response", "status": "failed", "statusDetails": {"message": "AssertionError: An unknown error occured", "trace": "get_job_details = ((['Product Manager', 'Mecanic Auto', 'Finisor Mobilier', 'Senior Angular Developer', 'Specialist Administrativ Operat...r', 'anahr', 'anahr', 'anahr', 'anahr', ...], ['on-site', 'on-site', 'on-site', 'on-site', 'on-site', 'on-site', ...]))\n\n    @pytest.mark.regression\n    @pytest.mark.API\n    def test_anahr_country_api(get_job_details):\n        allure.dynamic.title(f\"Test job countries from the {company_name} website against Peviitor API Response\")\n    \n        scraped_jobs_data, peviitor_jobs_data = get_job_details\n        with allure.step(\"Step 1: Get job countries and titles from the scraper\"):\n            job_countries_scraper = scraped_jobs_data[2]\n            job_titles_scraper = scraped_jobs_data[0]\n    \n        with allure.step(\"Step 2: Get job countries and titles from the Peviitor API\"):\n            job_countries_peviitor = peviitor_jobs_data[2]\n            job_titles_peviitor = peviitor_jobs_data[0]\n    \n        with allure.step(\"Step 3: Compare job countries from scraper response against Peviitor API Response\"):\n            allure.attach(f\"Expected Results: {job_countries_scraper}\", name=\"Expected Results\")\n            allure.attach(f\"Actual Results: {job_countries_peviitor}\", name=\"Actual Results\")\n>           TestUtils().check_job_countries(job_countries_scraper,job_titles_scraper, job_countries_peviitor, job_titles_peviitor)\n\ntests/Test_API_Suite/test_anahr.py:75: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <tests.utils.TestUtils object at 0x7fbba7d1b940>\nexpected_countries = ['Romania', 'Romania', 'Romania', 'Romania', 'Romania', 'Romania', ...]\nactual_countries = ['Product Manager', 'Mecanic Auto', 'Finisor Mobilier', 'Senior Angular Developer', 'Specialist Administrativ Operational', 'Junior Frontend Developer with Angular', ...]\njob_titles_scraper = ['Romania', 'Romania', 'Romania', 'Romania', 'Romania', 'Romania', ...]\napi_job_titles = ['Product Manager', 'Mecanic Auto', 'Finisor Mobilier', 'Senior Angular Developer', 'Specialist Administrativ Operational', 'Junior Frontend Developer with Angular', ...]\n\n    def check_job_countries(self, expected_countries, actual_countries, job_titles_scraper, api_job_titles):\n        if not expected_countries:\n            msg = f\"Scraper is not grabbing any job countries\"\n            allure.step(msg)\n            raise AssertionError(msg)\n    \n        msg = \"An unknown error occured\"\n    \n        # Check job countries from scraper against the peviitor api\n        scraper_actual_countries, scraper_job_titles = self.get_different_items(expected_countries, actual_countries, job_titles_scraper)\n    \n        if scraper_actual_countries:\n            msg = f\"Peviitor is missing job countries for the following job titles: {scraper_job_titles}\"\n            allure.step(msg)\n            raise AssertionError(msg)\n        else:\n            # Check job countries from peviitor against the scraper response\n            peviitor_actual_countries, peviitor_job_titles = self.get_different_items(actual_countries, expected_countries, api_job_titles)\n            if peviitor_actual_countries:\n                msg = f\"Peviitor is having extra job countries for the following titles: {peviitor_job_titles}\"\n                allure.step(msg)\n                raise AssertionError(msg)\n    \n        allure.step(msg)\n>       assert expected_countries == actual_countries, msg\nE       AssertionError: An unknown error occured\n\ntests/utils.py:178: AssertionError"}, "steps": [{"name": "Step 1: Get job countries and titles from the scraper", "status": "passed", "start": 1711861789388, "stop": 1711861789388}, {"name": "Step 2: Get job countries and titles from the Peviitor API", "status": "passed", "start": 1711861789388, "stop": 1711861789388}, {"name": "Step 3: Compare job countries from scraper response against Peviitor API Response", "status": "failed", "statusDetails": {"message": "AssertionError: An unknown error occured\n", "trace": "  File \"/home/runner/work/RaresTestHub/RaresTestHub/tests/Test_API_Suite/test_anahr.py\", line 75, in test_anahr_country_api\n    TestUtils().check_job_countries(job_countries_scraper,job_titles_scraper, job_countries_peviitor, job_titles_peviitor)\n  File \"/home/runner/work/RaresTestHub/RaresTestHub/tests/utils.py\", line 178, in check_job_countries\n    assert expected_countries == actual_countries, msg\n"}, "attachments": [{"name": "Expected Results", "source": "91395c52-6a90-4fe9-8d07-4d9213cc6341-attachment.attach"}, {"name": "Actual Results", "source": "4b0f319f-0192-48d3-b09d-74add9fa27f0-attachment.attach"}], "start": 1711861789388, "stop": 1711861789388}], "start": 1711861789387, "stop": 1711861789388, "uuid": "5e4b36c6-fa67-4522-b61e-3bf31ffeacde", "historyId": "50937ac439f5f8d053d6fa34f5434bd9", "testCaseId": "50937ac439f5f8d053d6fa34f5434bd9", "fullName": "Test_API_Suite.test_anahr#test_anahr_country_api", "labels": [{"name": "tag", "value": "API"}, {"name": "tag", "value": "regression"}, {"name": "parentSuite", "value": "Test_API_Suite"}, {"name": "suite", "value": "test_anahr"}, {"name": "host", "value": "fv-az771-813"}, {"name": "thread", "value": "2202-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Test_API_Suite.test_anahr"}]}